qark/plugins/intent/implicit_intent_to_pending_intent.py

```diff
@@ -0,0 +1,92 @@
+from qark.scanner.plugin import BasePlugin
+from qark.issue import Severity, Issue
+
+import logging
+import os
+import re
+
+import javalang
+from javalang.tree import MethodInvocation, ClassCreator, ReferenceType
+
+log = logging.getLogger(__name__)
+
+
+PENDING_INTENT_METHODS = ("getActivities",
+                          "getService",
+                          "getActivity",
+                          "getBroadcast",
+                          )
+
+
+class ImplicitIntentToPendingIntent(BasePlugin):
+    """
+    This plugin checks if a `new Intent` is passed into any of the `PENDING_INTENT_METHODS`.
+    If there is, a Vulnerability is created.
+    """
+    def __init__(self):
+        BasePlugin.__init__(self, category="intent", name="Empty pending intent found",
+                            description=("For security reasons, the Intent you supply here should almost always"
+                                         " be an explicit intent that is specify an explicit component to be delivered"
+                                         " to through Intent.setClass. A malicious application could potentially"
+                                         " intercept, redirect and/or modify this Intent. Pending Intents retain the"
+                                         " UID of your application and all related permissions, allowing another"
+                                         " application to act as yours. Reference: "
+                                         "https://developer.android.com/reference/android/app/PendingIntent.html"))
+        self.severity = Severity.VULNERABILITY
+        self.current_file = None
+
+    def run(self, files, apk_constants=None):
+        java_files = (decompiled_file for decompiled_file in files
+                      if os.path.splitext(decompiled_file.lower())[1] == ".java")
```

Comment:
```
>>> os.path.splitext('myfile.JAVA')
('myfile', '.JAVA')
```

We use the word `decompiled_file` here but it could be someones source code as well (greater chance of having stranger extensions).